tests/integration_tests/tests/agentless_tests/test_status_reporter.py

```diff
@@ -0,0 +1,92 @@
+#########
+# Copyright (c) 2017-2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+import json
+from time import sleep
+from uuid import uuid4
+
+from cloudify.constants import MANAGER_STATUS_REPORTER
+
+from integration_tests import AgentlessTestCase
+from integration_tests.framework import postgresql
+from integration_tests.tests.utils import get_resource
+from integration_tests.tests.constants import MANAGER_PYTHON
+
+
+class TestManagerStatusReporter(AgentlessTestCase):
+    def test_status_reporter_has_correct_key_after_snapshot_restore(self):
+        snapshot_id = "test_" + str(uuid4())
+        initial_token_key, _ = self.get_reporter_token_key()
+
+        execution = self.client.snapshots.create(snapshot_id, False)
+        self.wait_for_execution_to_end(execution)
+
+        self.update_reporter_token_key(uuid4().hex)
```

Comment:
you can also just use a const string and not uuid4

Reply:
Changed to contain 32 `a` letters